src/Microsoft.FeatureManagement/IFeatureVariantOptionsResolver.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.FeatureManagement
+{
+    /// <summary>
+    /// Performs the resolution and binding necessary in the feature variant resolution process. 
+    /// </summary>
+    public interface IFeatureVariantOptionsResolver
+    {
+        /// <summary>
+        /// Retrieves typed options for a given feature definition and chosen variant.
+        /// </summary>
+        /// <typeparam name="T">The type of the options to return.</typeparam>
+        /// <param name="featureDefinition">The definition of the feature that the resolution is being performed for.</param>
+        /// <param name="variant">The chosen variant of the feature.</param>
+        /// <param name="cancellationToken">The cancellation token to cancel the operation.</param>
+        /// <returns>Typed options for a given feature definition and chosen variant.</returns>
+        ValueTask<T> GetOptions<T>(FeatureDefinition featureDefinition, FeatureVariant variant, CancellationToken cancellationToken);
```

Comment:
>FeatureDefinition featureDefinition [](http://example.com/codeflow?start=35&length=35)

In which scenario could options resolver need to know the feature definition?

Reply:
With the built-in solution we only need the `FeatureVariant` because we only use the configuration reference to lookup in the application's config and resolve the variant's options. However, it is entirely possible that a variant's options get pulled from a server. In this case it is reasonable that the feature information that the variant belongs to may be required to complete the request.